cmd/zc_traverser_blob.go

```diff
@@ -52,29 +53,45 @@ func (t *blobTraverser) isDirectory(isSource bool) bool {
 		return isDirDirect
 	}
 
-	_, isSingleBlob := t.getPropertiesIfSingleBlob()
+	_, isSingleBlob, err := t.getPropertiesIfSingleBlob()
+
+	if stgErr, ok := err.(azblob.StorageError); ok {
+		// We know for sure this is a single blob still, let it walk on through to the traverser.
+		if stgErr.ServiceCode() == common.CPK_ERROR_SERVICE_CODE {
+			return true
```

Comment:
Returning true here seems strange to me.  E.g. if the blob is the source of a copy, and user has not used --recursive, won't this result in the display of the message "cannot use directory as source without --recursive or trailing wildcard (/*)"  That's confusing to the user. 

Intuitively it feels more correct to return false here, but I'm guessing from the comment there's some particular reason why true was chosen in this case.  What was that reason, and what about the issue with it displaying the "cannot use directory as a source" message?